plugins/module_utils/redfish_utils.py

```diff
@@ -3394,3 +3394,77 @@ def get_hpe_fan_percent_min(self):
             fan_percent_min_config = hpe.get('FanPercentMinimum')
         result["fan_percent_min"] = fan_percent_min_config
         return result
+
+    def create_volume(self, volume_details):
+        # Find the Storage resource from the requested ComputerSystem resource
+        response = self.get_request(self.root_uri + self.systems_uri)
+        if response['ret'] is False:
+            return response
+        data = response['data']
+        storage_uri = data.get('Storage', {}).get('@odata.id')
+        if storage_uri is None:
+            return {'ret': False, 'msg': 'Storage resource not found'}
+
+        # Get Storage Collection
+        response = self.get_request(self.root_uri + storage_uri)
+        if response['ret'] is False:
+            return response
+        data = response['data']
+
+        # Collect Storage Controllers
+        self.storage_controlllers_uris = [i['@odata.id'] for i in response['data'].get('Members', [])]
+        if not self.storage_controlllers_uris:
+            return {
+                'ret': False,
+                'msg': "StorageCollection's Members array is either empty or missing"}
+
+        # Check if storage controller supports volume creation
+        response = self.get_request(self.root_uri + self.storage_controlllers_uris[0])
+        if response['ret'] is False:
+            return response
+        data = response['data']
+
+        if data.get('Volumes'):
+            response = self.get_request(self.root_uri + data['Volumes']['@odata.id'])
+            if response['ret'] is False:
+                return response
+            data = response['data']
+
+            if data.get('@Redfish.CollectionCapabilities') and data['@Redfish.CollectionCapabilities'].get('Capabilities') \
```

Comment:
Showing `@Redfish.CollectionCapabilities` is not an indicator if the collection supports creating Volumes. The term is used to help clients learn what properties are supported in the POST request. Since we're not stepping into the Capabilities object for performing anything to fine-tune the request, I would recommend removing this.

Reply:
Removing as suggested.

Reply:
Resolving this conversation. Thanks!